docs/spec/generics.rst

```diff
@@ -458,8 +450,43 @@ Also consider the following example::
   class MyDict(Mapping[str, T]):
       ...
 
-In this case MyDict has a single parameter, T.
+In this case ``MyDict`` has a single parameter, ``T``.
+
+Type variables are applied to the defined class in the order in which
+they first appear in any generic base classes::
+
+  from typing import Generic, TypeVar
+
+  T1 = TypeVar('T1')
+  T2 = TypeVar('T2')
+  T3 = TypeVar('T3')
+
+  class Parent1(Generic[T1, T2]):
+      ...
+  class Parent2(Generic[T1, T2]):
+      ...
+  class Child(Parent1[T1, T3], Parent2[T2, T3]):
+      ...
+
+That ``Child`` definition is equivalent to::
+
+  class Child(Parent1[T1, T3], Parent2[T2, T3], Generic[T1, T3, T2]):
+      ...
 
+Type checkers may warn when the type variable order is inconsistent::
```

Comment:
Should we mandate an error here instead? This code doesn't make sense, because a `Child[int, str]` is both a `Parent[int, str]` and a `Parent[str, int]` due to the double inheritance. So a type checker should error here, because it can't build up this class in a correct way.

Reply:
The reason I didn't mandate an error is that mypy currently doesn't error on this. But I'm happy to change it if we think that would be better.

Reply:
I don't know if there is a rationale behind mypy's behavior, or how hard it would be to change, but it seems to me that this should be mandated as an error; I don't see any way to make sense of this definition.

> This code doesn't make sense, because a `Child[int, str]` is both a `Parent[int, str]` and a `Parent[str, int]` due to the double inheritance.

To make sure I'm understanding: you meant `Grandparent` here? There is no double inheritance from `Parent`.

Reply:
Ok, I updated this section to mandate an error.